File: 151004/Bosko/lab1/lab1/DTO/NoteResponseTo.cs
using lab1.DTO.Interface;
using lab1.Models;

namespace lab1.DTO
{
    public class NoteResponseTo(int Id, int? NewsId, string? Content, News? News) : IResponseTo
    {
        public int Id { get; set; } = Id;

        public int? NewsId { get; set; } = NewsId;

        public string? Content { get; set; } = Content;

        public News? News { get; set; } = News;
    }
}

File: 151002/Strukov/REST News/discussion/src/main/java/by/bsuir/news/dto/response/NoteResponseTo.java
package by.bsuir.news.dto.response;

import by.bsuir.news.entity.Note;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class NoteResponseTo {
    private Long id;
    private String content;
    private Long newsId;

    public static NoteResponseTo toResponse(Note source) {
        NoteResponseTo model = new NoteResponseTo();
        model.id = source.getKey().getId();
        model.content = source.getContent();
        model.newsId = source.getKey().getNewsId();
        return model;
    }
}
